src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java

```diff
@@ -1,173 +1,261 @@
 
 @Service
-public class OrdersKafkaConsumer implements InitializingBean {
+public class OrdersKafkaConsumer implements ConsumerSeekAware {
     private static final Logger LOGGER = LoggerFactory.getLogger(APPLICATION_NAMESPACE);
     private static final String ORDER_RECEIVED_TOPIC = "order-received";
     private static final String ORDER_RECEIVED_TOPIC_RETRY = "order-received-retry";
+    private static final String ORDER_RECEIVED_KEY_RETRY = ORDER_RECEIVED_TOPIC_RETRY;
     private static final String ORDER_RECEIVED_TOPIC_ERROR = "order-received-error";
     private static final String ORDER_RECEIVED_GROUP = APPLICATION_NAMESPACE + "-" + ORDER_RECEIVED_TOPIC;
     private static final String ORDER_RECEIVED_GROUP_RETRY = APPLICATION_NAMESPACE + "-" + ORDER_RECEIVED_TOPIC_RETRY;
     private static final String ORDER_RECEIVED_GROUP_ERROR = APPLICATION_NAMESPACE + "-" + ORDER_RECEIVED_TOPIC_ERROR;
-    private static final int MAX_RETRY_ATTEMPTS = 3;
+    private static long errorRecoveryOffset = 0l;
 
-    private CHKafkaResilientConsumerGroup chKafkaConsumerGroupMain;
-    private CHKafkaResilientConsumerGroup chKafkaConsumerGroupRetry;
     @Value("${spring.kafka.consumer.bootstrap-servers}")
-    private String brokerAddresses;
+    private String bootstrapServers;
+    @Value("${uk.gov.companieshouse.item-handler.error-consumer}")
+    private boolean errorConsumerEnabled;
     private final SerializerFactory serializerFactory;
+    private final OrdersKafkaProducer kafkaProducer;
+    private final KafkaListenerEndpointRegistry registry;
 
-    public OrdersKafkaConsumer(SerializerFactory serializerFactory) {
+    public OrdersKafkaConsumer(SerializerFactory serializerFactory,
+                               OrdersKafkaProducer kafkaProducer,
+                               KafkaListenerEndpointRegistry registry) {
         this.serializerFactory = serializerFactory;
+        this.kafkaProducer = kafkaProducer;
+        this.registry = registry;
     }
 
-    @Override
-    public void afterPropertiesSet() throws Exception {
-        LOGGER.debug("Initializing kafka consumer service " + this.toString());
-
-        ProducerConfig config = getProducerConfig();
-        CHKafkaProducer chKafkaProducer = new CHKafkaProducer(config);
-
-        ConsumerConfig consumerConfig = getConsumerConfig();
-
-        consumerConfig.setGroupName(ORDER_RECEIVED_GROUP);
-        consumerConfig.setTopics(Collections.singletonList(ORDER_RECEIVED_TOPIC));
-        chKafkaConsumerGroupMain
-                = new CHKafkaResilientConsumerGroup(consumerConfig, CHConsumerType.MAIN_CONSUMER,
-                        new KafkaConsumerFactory(), chKafkaProducer);
+    /**
+     * Main listener/consumer. Calls a `retryable` method to process received message.
+     * If the `retryable` processor is unsuccessful with a `retryable` error, after maximum numbers of attempts allowed,
```

Comment:
> Calls a `retryable` method to process received message.

Perhaps this remark now really relates to the `handleMessage` method?

Reply:
Fixed.